src/base/utils/gzip.cpp

```diff
@@ -64,39 +62,46 @@ QByteArray Utils::Gzip::compress(const QByteArray &data, const int level, bool *
     if (result != Z_OK)
         return {};
 
-    QByteArray output;
-    output.reserve(deflateBound(&strm, data.size()));
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 5, 0))
+    QByteArray ret {static_cast<qsizetype>(deflateBound(&strm, data.size())), Qt::Uninitialized};
+#else
+    QByteArray ret {static_cast<int>(deflateBound(&strm, data.size())), Qt::Uninitialized};
+#endif
+    strm.next_out = reinterpret_cast<Bytef *>(ret.data());
+    strm.avail_out = ret.size();
 
     // feed to deflate
     while (strm.avail_in > 0)
     {
-        result = deflate(&strm, Z_NO_FLUSH);
+        Q_ASSERT(strm.avail_out > 0);
 
-        if (result != Z_OK)
+        result = deflate(&strm, Z_NO_FLUSH);
```

Comment:
I wonder why isn't "single `deflate()` call" way is used as described in the gzip docs:
>Z_FINISH can be used in the first deflate call after deflateInit if all the compression is to be done in a single step. In order to complete in one call, avail_out must be at least the value returned by deflateBound (see below). Then deflate is guaranteed to return Z_STREAM_END.

Reply:
I was concerned it might have inferior compression rate but my test results says they are the same.

PR updated.